tests/Oryx.Integration.Tests/StorageAccountSanityTestBase.cs

```diff
@@ -198,9 +198,19 @@ private void AssertExpectedListOfBlobs(string platformName, params string[] expe
 
         private XDocument GetMetadata(string platformName)
         {
-            var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, _storageUrl, platformName);
+            var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, _storageUrl, platformName, string.Empty);
             var blobList = _httpClient.GetStringAsync(url).Result;
-            return XDocument.Parse(blobList);
+            XDocument xdoc = XDocument.Parse(blobList);
+            var marker = xdoc.Root.Element("NextMarker").Value;
+            do
+            {
+                url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, _storageUrl, platformName, marker);
+                var blobListFromNextMarker = _httpClient.GetStringAsync(url).Result;
+                marker = XDocument.Parse(blobListFromNextMarker).Root.Element("NextMarker").Value;
```

Comment:
nit: we are parsing the string to XDocument twice (here and next line). Let's pull that logic out so we only do it once